### PR TITLE
Workaround for https://github.com/LedgerHQ/app-ethereum/issues/409

### PR DESCRIPTION
**Pull request by prestwich**
makes chunk size dynamic to avoid running into this:
https://github.com/LedgerHQ/app-ethereum/issues/409

## Motivation

https://github.com/foundry-rs/foundry/issues/4362

## Solution

- find the highest chunk size <= 255 that will not trigger the bug
- drive-by: address some edge cases in payload handling that would have caused panics with incorrect ledger usage
- drive-by: add tracing to ledger app (WIP)

## PR Checklist

-   [ ] Added Tests
-   [x] Added Documentation
-   [x] Updated the changelog
-   [ n/a ] Breaking changes

### CODE REVIEW

**Comment by prestwich** [comment]
now ready for review 🖖

File: ethers-signers/src/ledger/app.rs [hunk 1]
```diff
@@ -204,25 +225,47 @@ impl LedgerEthereum {
             response_len: None,
         };
 
-        let mut result = Vec::new();
+        let mut answer = None;
+        // workaround for https://github.com/LedgerHQ/app-ethereum/issues/409
+        // TODO: remove in future version
+        let chunk_size =
+            (0..=255).rev().find(|i| payload.len() % i != 3).expect("true for any length");
```
Comment (prestwich) [inline ethers-signers/src/ledger/app.rs:232]: finds the largest chunk size that does not leave 3 hanging bytes on the payload lol

Comment (gakonst) [reply ethers-signers/src/ledger/app.rs:232]: ridiculous

File: ethers-signers/src/ledger/app.rs [hunk 2]
```diff
@@ -204,25 +225,47 @@ impl LedgerEthereum {
...
-            let data = payload.drain(0..chunk_size).collect::<Vec<_>>();
-            command.data = APDUData::new(&data);
-
-            let answer = block_on(transport.exchange(&command))?;
-            result = answer.data().ok_or(LedgerError::UnexpectedNullResponse)?.to_vec();
+        let span = tracing::debug_span!("send_loop", index = 0, chunk = "");
+        let guard = span.entered();
+        for (index, chunk) in payload.chunks(chunk_size).enumerate() {
+            guard.record("index", index);
+            guard.record("chunk", hex::encode(chunk));
```
Comment (gakonst) [inline ethers-signers/src/ledger/app.rs:239]: TIL

**Review by gakonst — APPROVED** [review]
